pages/playground/index.tsx

```diff
@@ -0,0 +1,324 @@
+import { useVisitorData } from '@fingerprintjs/fingerprintjs-pro-react';
+import { UseCaseWrapper } from '../../client/components/use-case-wrapper';
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Alert,
+  Box,
+  CircularProgress,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { useQuery } from 'react-query';
+import { useState } from 'react';
+import { CodeSnippet } from '../../client/components/CodeSnippet';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+
+import { green, red } from '@mui/material/colors';
+import { lightGreen } from '@mui/material/colors';
+import { blueGrey } from '@mui/material/colors';
+import dynamic from 'next/dynamic';
+import { useUserPreferences } from '../../client/api/personalization/use-user-preferences';
+import MyTable, { TableCellData } from '../../client/components/playground/MyTable';
+import { EventResponse } from '@fingerprintjs/fingerprintjs-pro-server-api';
+import BotDetectionResult from '../../client/components/playground/BotDetectionResult';
+import Info from '../../client/components/playground/InfoIcon';
+import RefreshButton from '../../client/components/playground/RefreshButton';
+
+// Map cannot be server-side rendered
+const Map = dynamic(() => import('../../client/components/playground/Map'), { ssr: false });
+
+function Playground() {
+  const {
+    data: agentResponse,
+    isLoading: isLoadingAgentResponse,
+    getData: getAgentData,
+    error: agentError,
+  } = useVisitorData({ extendedResult: true, ignoreCache: true }, { immediate: true });
+
+  const requestId = agentResponse?.requestId;
+
+  /** Temporary fix to store previous because ReactQuery sets data to undefined before the fresh data is available when I make a new query and it makes everything flash */
```

Comment:
So the bahavior I want is "fetch the server event and show me that data always until I fetch the new event, fetch new event when `requestId` changes". 

Problem with the current setup is the when the `requestId` changes, I guess a new key means it's a completely new query that is `undefined` until new data comes and that makes everything flash. I want to avoid that `undefined` flash. 

Another way I tried is specifying a constant query ket like "event" and invalidating that query with `useEffect(() => { invalidateQuery("event"), [requestId] }`, but that does not seem to work.

Am I missing something? Is there a better to achieve this with React Query? Any ideas @TheUnderScorer? Thanks a lot!

Reply:
Hmm, I tried to fix it locally but couldn't come up with anything clever. As you said, you use the `requestId` as query key, so whenever it changes the whole query is re-created, so I guess it's the best solution for now.